com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -487,21 +463,23 @@ static private ShaderDesc.Shader.Builder buildSpirvFromGLSL(ByteArrayInputStream
         return builder;
     }
 
-    public ShaderDesc compile(ByteArrayInputStream is, ES2ToES3Converter.ShaderType shaderType, IResource resource, String resourceOutput, String platform, boolean isDebug, boolean outputSpirv, boolean soft_fail) throws IOException, CompileExceptionError {
+    public ShaderDesc compile(ShaderIncludeCompiler includeCompiler, ES2ToES3Converter.ShaderType shaderType, IResource resource,
+        String resourceOutput, String platform, boolean isDebug, boolean outputSpirv, boolean soft_fail) throws IOException, CompileExceptionError {
         ShaderDesc.Builder shaderDescBuilder = ShaderDesc.newBuilder();
 
         // Build platform specific shader targets (e.g SPIRV, MSL, ..)
         Platform platformKey = Platform.get(platform);
         if(platformKey != null) {
+            String shaderSource = includeCompiler.getCompiledSource();
```

Comment:
Maybe "include compiler" isn't the best name for this 🤔 we use the compiler to generate the source from the actual input GLSL, so perhaps "ShaderSourceGenerator" or something like that?